_posts/2025-10-21-micrometer-prometheus-v1.adoc

```diff
@@ -0,0 +1,83 @@
+
+[INFO]
+====
+Users should move to the new `quarkus-micrometer-registry-prometheus-v1` extension.
+====
+
+Since the new client introduces *breaking changes*, we’ve prepared a migration plan designed to reduce the impact on existing users.
+
+=== Migration plan
+
+1. Keep current extension as is until Quarkus 4.0 (whenever that happens) so users can migrate at a time of their choice.
+2. Introduce a new extension called https://docs.quarkiverse.io/quarkus-micrometer-registry/dev/micrometer-registry-prometheus-v1.html[`quarkus-micrometer-registry-prometheus-v1`].
+3. Remove current `quarkus-micrometer-registry-prometheus` on Quarkus 4.0 (no timeline at this moment) while providing an alternative on Quarkiverse. Quarkus users will get an additional migration period by using an upcoming community supported Quarkiverse extension.
+4. Refactor code on Quarkus 4.0 Core:
+** Move the deprecated `quarkus-micrometer-registry-prometheus` extension from Quarkus core repo to Quarkiverse.
+** Move the new https://docs.quarkiverse.io/quarkus-micrometer-registry/dev/micrometer-registry-prometheus-v1.html[`quarkus-micrometer-registry-prometheus-v1`] extension from Quarkiverse to Quarkus core repo and make it the default registry.
+** Refactor the https://quarkus.io/guides/telemetry-micrometer[`quarkus-micrometer`] extension tests to use the new client.
+
+=== Prometheus Client v1.x changes
+
+The `quarkus-micrometer-registry-prometheus-v1` extension will be using the Prometheus Client v1.x and includes *breaking changes* expressed in detail on the https://github.com/micrometer-metrics/micrometer/wiki/1.13-Migration-Guide[Micrometer 1.13 Migration Guide], featuring package, API and semantic convention changes.
+
+These are the main points to consider when migrating to the new Quarkus extension:
+
+* Counters now use Longs instead of Doubles.
+** Before: `"registry=\"prometheus\",status=\"200\",uri=\"/example/prime/{number}\"} 2.0`
+** Now: `"registry=\"prometheus\",status=\"200\",uri=\"/example/prime/{number}\"} 2`
+* `io.micrometer.prometheusmetrics.PrometheusMeterRegistry` must be used instead of the old `io.micrometer.prometheus.PrometheusMeterRegistry`.
+* `io.prometheus.metrics.tracer.common.SpanContext` must be used instead of the old `io.prometheus.client.exemplars.ExemplarSampler`
+* The new Prometheus client uses some reserved words that must not be used to name metrics, therefore some metrics were renamed. Some examples of reserved words and renamed metrics:
+** `info`. Before: `jvm_info_total`. Now: `jvm_total`
+** `duration`. Before: `http_server_requests_duration_seconds`. Now: `http_server_requests_seconds`
+* Some metrics would display Tags ending in a comma (`,`). This tailing comma has now been removed. Example: `"hibernate_flushes_total{entityManagerFactory=\"<default>\",env=\"test\",env2=\"test\",registry=\"prometheus\",} 1.0"`
```

Comment:
Why do we have the \" ?

Reply:
It was just to delimit the String and it can go away.